VocaDb.ResXFileCodeGenerator/FileOptions.cs

```diff
@@ -0,0 +1,118 @@
+	{
+		File = file;
+		var resxFilePath = file.Path;
+		LocalNamespace = Utilities.GetLocalNamespace(
+			resxFilePath,
+			options.TryGetValue("build_metadata.EmbeddedResource.TargetPath", out var targetPath) &&
+			targetPath is { Length: > 0 }
+				? targetPath
+				: null,
+			globalOptions.ProjectFullPath,
+			globalOptions.RootNamespace);
+
+		CustomToolNamespace =
+			options.TryGetValue("build_metadata.EmbeddedResource.CustomToolNamespace",
+				out var customToolNamespace) && customToolNamespace is { Length: > 0 }
+				? customToolNamespace
+				: null;
+
+		ClassName = Utilities.GetClassNameFromPath(resxFilePath);
+
+		NullForgivingOperators = globalOptions.NullForgivingOperators;
+
+		PublicClass =
+			options.TryGetValue("build_metadata.EmbeddedResource.PublicClass", out var perFilePublicClassSwitch) &&
+			perFilePublicClassSwitch is { Length: > 0 }
+				? perFilePublicClassSwitch.Equals("true", StringComparison.OrdinalIgnoreCase)
+				: globalOptions.PublicClass;
+
+		StaticClass =
+			options.TryGetValue("build_metadata.EmbeddedResource.StaticClass", out var perFileStaticClassSwitch) &&
+			perFileStaticClassSwitch is { Length: > 0 }
+				? !perFileStaticClassSwitch.Equals("false", StringComparison.OrdinalIgnoreCase)
+				: globalOptions.StaticClass;
+
+		StaticMembers =
+			options.TryGetValue("build_metadata.EmbeddedResource.StaticMembers", out var staticMembersSwitch) &&
+			staticMembersSwitch is { Length: > 0 }
+				? !staticMembersSwitch.Equals("false", StringComparison.OrdinalIgnoreCase)
+				: globalOptions.StaticMembers;
+
+		PartialClass =
+			options.TryGetValue("build_metadata.EmbeddedResource.PartialClass", out var partialClassSwitch) &&
+			partialClassSwitch is { Length: > 0 }
+				? partialClassSwitch.Equals("true", StringComparison.OrdinalIgnoreCase)
+				: globalOptions.PartialClass;
+
+		InnerClassVisibility = globalOptions.InnerClassVisibility;
+		if (options.TryGetValue("build_metadata.EmbeddedResource.InnerClassVisibility",
+			    out var innerClassVisibilitySwitch) &&
+		    Enum.TryParse(innerClassVisibilitySwitch, true, out InnerClassVisibility v) &&
+		    v != InnerClassVisibility.SameAsOuter)
+		{
+			InnerClassVisibility = v;
+		}
+
+		InnerClassName = globalOptions.InnerClassName;
+		if (options.TryGetValue("build_metadata.EmbeddedResource.InnerClassName", out var innerClassNameSwitch) &&
+		    innerClassNameSwitch is { Length: > 0 })
+		{
+			InnerClassName = innerClassNameSwitch;
+		}
+
+		InnerClassInstanceName = globalOptions.InnerClassInstanceName;
+		if (options.TryGetValue("build_metadata.EmbeddedResource.InnerClassInstanceName", out var innerClassInstanceNameSwitch)&&
+		    innerClassInstanceNameSwitch is { Length: > 0 })
+		{
+			InnerClassInstanceName = innerClassInstanceNameSwitch;
+		}
+
+		FilePath = resxFilePath;
+
+		Valid = globalOptions.Valid;
+	}
+
+	public static FileOptions Select(AdditionalText file, AnalyzerConfigOptionsProvider options,
```

Comment:
For your information, you can make it internal by using `InternalsVisibleTo`.

Reply:
I know, doesn't matter in this case since its not a library anyone is ever expected to include